lib/services/check_quantity.rb

```diff
@@ -0,0 +1,9 @@
+module Store
+  class CheckQuantity
+    def call(quantity)
+      raise InvalidQuantityError unless quantity
+      raise InvalidQuantityError unless quantity.is_a?(Integer)
+      raise InvalidQuantityError if quantity < 0
```

Comment:
looks like it can be a method. If you use it more then one (in more classes) you can extract it to module and include